<?xml version="1.0"?>
<launch>
	<!-- Global Arguments -->
	<arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
	<arg name="lidar" default="true"/>
	<arg name="slam" default="true"/>
	<arg name="fdet" default="true"/>

	<!-- Arguments: LIDAR -->
	<arg name="port" default="/dev/ttyUSB0" />
	<arg name="frame_id" default="map" />

	<!-- Arguments: SLAM -->
	<arg name="slam_methods" default="gmapping" doc="slam type [gmapping, cartographer, hector, karto, frontier_exploration]"/>
	<arg name="configuration_basename" default="turtlebot3_lds_2d.lua"/>

	<!-- Drivers: LIDAR -->
  	<group if="$(arg lidar)">
		<include file="$(find hls_lfcd_lds_driver)/launch/hlds_laser.launch" />
	</group>

	<!-- Drivers: SLAM -->
	<group if="$(arg slam)">
	  	<!-- TurtleBot3 -->
		<include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
    		<arg name="model" value="$(arg model)" />
  		</include>

		<!-- SLAM: Gmapping, Cartographer, Hector, Karto, Frontier_exploration, RTAB-Map -->
		<include file="$(find turtlebot3_slam)/launch/turtlebot3_$(arg slam_methods).launch">
			<arg name="model" value="$(arg model)"/>
			<arg name="configuration_basename" value="$(arg configuration_basename)"/>
		</include>
	</group>

	<node pkg="robofriend" type="RobobrainNode.py" name="robofriend_robobrain" output="screen" />
	<node pkg="robofriend" type="GameCommunicator.py" name="robofriend_game_communicator" output="screen" />
	<node pkg="robofriend" type="PCBSensorData.py" name="robofriend_pcb_sensor_data" output="screen" />
	<node pkg="robofriend" type="RFIDReader.py" name="robofriend_rfid_reader" output="screen" />
	<node pkg="robofriend" type="TeensyCommunicator.py" name="robofriend_teensy_communicator" output="screen" />
	<node pkg="robofriend" type="Webserver.py" name="robofriend_web_server" output="screen" />
	<group if="$(arg fdet)">
		<node pkg="robofriend" type="FaceDetectionNode.py" name="robofriend_face_detection_node" output="screen" />
		<node pkg="robofriend" type="IOWarriorNode.py" name="robofriend_io_warrior_handler" output="screen" />
		<node pkg="robofriend" type="KeyboardNode.py" name="robofriend_keyboard_node" output="screen" />
		<node pkg="robofriend" type="SpeechNode.py" name="robofriend_speech_node" output="screen" />
		<node pkg="robofriend" type="ServoCamNode.py" name="robofriend_servo_cam_node" output="screen" />
		<node pkg="robofriend" type="LedEarsNode.py" name="robofriend_led_ears" output="screen" />
	</group>
</launch>
